test/src/main/java/org/apache/accumulo/test/shell/ShellServerIT.java

```diff
@@ -1841,8 +1845,8 @@ public void scansWithNeverHostedTablets() throws Exception {
     final String table = getUniqueNames(1)[0];
     ts.exec("createtable " + table);
     ts.exec("addsplits -t " + table + " a c e g m t");
-    ts.exec("goal -t " + table + " -b a -e a -g never");
-    ts.exec("goal -t " + table + " -b c -e e -ee -g always");
+    ts.exec("setgoal -t " + table + " -b a -e a -g never");
```

Comment:
I think `setgoal` is not a very good name for a tablet-specific function, as it makes one wonder "goal for what?", because "goal" is too generic of a term.

I think "hosttablet ... always/never/ondemand" or "settabletgoal ... always/never/ondemand". Even these are probably not the best names either, but they give a lot more context to what kind of goal is being set.

Alternatively, one could have a command named "tablet" and a sub-command called something like "online", so you get a very expressive command like: "tablet online... always/never/ondemand" or maybe "tablet hosted... always/never/ondemand".

Or we could overload the current `online` command to be able to set a range, and an optional flag for `ondemand`. So, you get the following:

```sh
> online -t table -b beginRange -e endRange             # "always"
> offline -t table -b beginRange -e endRange            # "never"
> online -t table -b beginRange -e endRange --ondemand  # "ondemand"
```

Reply:
Elsewhere, we call this "hosting goal", so "sethostinggoal" would be appropriate as well.

Reply:
@ctubbsii  the name of that command is out of scope for this PR, but what are you wrote sounds reasonable.  Can you open an issue for reconsidering the name?

Reply:
Opened #3800